lib/connection.js

```diff
@@ -44,7 +45,7 @@ Connection.prototype.connect = function (timeout) {
         new Promise(function (resolve, reject) {
             setTimeout(function () {
                 reject(new NoKafkaConnectionError(self.server(), 'Connection timeout'));
-            }, timeout || 3000);
+            }, timeout || self.timeout);
```

Comment:
If you are connecting from a cluster in a different region, a longer timeout is very helpful.